src/kernel/include/hart_locals.h

```diff
@@ -12,6 +12,16 @@ struct hart_locals {
    * The ID of the hart. Not guaranteed to be small or densely packed.
    */
   u64 hart_id;
+  
+  /**
+   * Number of times interrupts have been disabled.
+   */
+  u32 noff;
+
+  /**
+   * Were interrrupts enabled before noff nesting?
+   */
+  u32 intena;
```

Comment:
more descriptive variable names? we don't need to worry about "the compiler only lets you use 6 letters in a variable name," unlike unix of old